pkg/ottl/functions.go

```diff
@@ -47,42 +47,57 @@ func (p *Parser[K]) newFunctionCall(ed editor) (Expr[K], error) {
 	return Expr[K]{exprFunc: fn}, err
 }
 
+func checkForArgErrors(index int, argsVal reflect.Value) (int, error) {
```

Comment:
```suggestion
func getArgumentIndex(index int, args reflect.Value) (int, error) {
```
We should rename this to clarify that we're looking to get the argument's index from the `ottlarg` tag. Also I think renaming the second parameter to `args` makes it more clear we're talking about the `Arguments` struct here, as opposed to a particular value. I think it's normally called `argsVal` because it's a `reflect.Value` instead of an `Arguments`-typed value.